src/common/HipDataUtils.hpp

```diff
@@ -10,73 +10,69 @@
 /// Methods for HIP kernel data allocation, initialization, and deallocation.
 ///
 
```

Comment:
What's up with all the formatting changes in this file?

Reply:
Ah I must of ran clang-format on the files I modified.... We can revert this

Reply:
Actually what do you guys think of running clang-format, in general?

Reply:
In my experience, clang-format really screws up the formatting for complex template expressions making them much less readable. That's why I made a script for RAJA (in the scripts directory) that filters out the examples, exercises, etc. Here, the kernels rarely change after they are added. So I would prefer that the author of a kernel formats it in a way that makes it most readable. Then, don't change the formatting after that.

Reply:
If possible, it would be nice to get a config such that we can use clang-format more widely than we do (can't say whether the specific issue is one with a fix yet, but if it is that would be nice).  Otherwise, can we start getting things that shouldn't be formatted by it marked appropriately with `//nofmt` comments so using it on the file as a whole doesn't mess things up?  It's a more useful tool if we don't have to think about it and can just always use it than if it has to be applied only in certain cases.

Reply:
I agree for the most part. But sprinkling special comments all over the code is also a pain. In Axom we use clang-format only with clang10 and later. There was a substantial expansion of configuration options with the 10 release of the clang toolchain. I haven't had a chance to look into that with RAJA code. It may be workable, needing only a small number of format filtering comments in the code.

Reply:
Yeah, the comments can be a pain (especially if there have to be a lot of them).  It's something that really only works if they can be used sparingly.

Setting a minimum clang-format version (higher than the required compiler version) seems completely reasonable to me.  It's best to set that to something anyway so people get consistent results.  Maybe something we can try and see how it goes.

Reply:
If we decide to use clang-format as part of our regular development process, we should add a check for it in our azure CI.

Reply:
Actually, this one looks like it had clang-format run on it *without the RAJA config for clang-format*. That's probably a non-trivial part of the difference.  Either that or some other formatter.  We have an explicit option set to avoid argument bin-packing and some other stuff that's happening here.

Reply:
Also, we should use clang-format 12.  They finally added `IndentPragmas`!

Reply:
I'll work on updating the clang-format config file to include more options it supports that we find useful in other projects. I'll put up a PR and we can debate it.